src/tdlib/chat_history.rs

```diff
@@ -123,6 +125,11 @@ impl ChatHistory {
         imp.is_loading.set(false);
 
         let enums::Messages::Messages(data) = result.map_err(ChatHistoryError::Tdlib)?;
+
+        if data.messages.is_empty() {
+            return Err(ChatHistoryError::EndOfChatReached);
```

Comment:
IMHO, this should not be encoded as an error, as you are expected to reach the end of the chat at some point. I would rather change the return type to `Result<bool, ChatHistoryError>` which indicates if more messages can be loaded.

Reply:
Right, it makes more sense this way.